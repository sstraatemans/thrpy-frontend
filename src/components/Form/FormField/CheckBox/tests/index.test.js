import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { UnWrapped } from "./../";

describe("<StyledCheckBox />", () => {
  it("render component <UnWrapped>", () => {
    let props = {
      name: "terms",
      val: true,
      onChange: () => {}
    };
    const wrapper = mount(<UnWrapped {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("sets checked", () => {
    let props = {
      id: "terms",
      val: true,
      onChange: () => {}
    };
    const renderedComponent = mount(<UnWrapped {...props} />);
    expect(renderedComponent.find("input").prop("checked")).toEqual("checked");
  });
});
