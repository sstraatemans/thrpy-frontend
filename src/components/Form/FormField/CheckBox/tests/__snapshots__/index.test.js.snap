exports[`<StyledCheckBox /> render component <UnWrapped> 1`] = `
<StyledCheckBox
  name="terms"
  onChange={[Function]}
  val={true}>
  <styled.div>
    <div
      className="sc-bxivhb eyxYjs">
      <styled.label
        htmlFor="terms"
        validationError="">
        <label
          className="sc-bZQynM jYCBXE"
          htmlFor="terms">
          <styled.input
            checked="checked"
            id="terms"
            name="terms"
            onChange={[Function]}
            type="checkbox">
            <input
              checked="checked"
              className="sc-ifAKCX gcmtRL"
              id="terms"
              name="terms"
              onChange={[Function]}
              type="checkbox" />
          </styled.input>
          <styled.span
            checked="checked">
            <span
              checked="checked"
              className="sc-EHOje hyuYGb" />
          </styled.span>
        </label>
      </styled.label>
    </div>
  </styled.div>
</StyledCheckBox>
`;
